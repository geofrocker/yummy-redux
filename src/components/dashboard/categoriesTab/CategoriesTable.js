import React, { Component } from 'react';
import { connect } from 'react-redux';
import { url } from '../../../config';
import { loadCategories, addCategory, loadCategory, updateCategory, deleteCategory} from '../../../actions/categoriesAction';
import { loadRecipes } from '../../../actions/recipesActions'
import { Table } from 'react-bootstrap';
import Pagination from './Pagination';
import CategoryTableRows from './CategoryTableRows';
import CategoryModal from './CategoryModal'
import toastr from 'toastr'

export class CategoriesTable extends Component {
  constructor(props) {
    super();
    this.state = {
      categories:[],
      showMessage: false,
      q: '',
      show: false,
      selected: '',
      modalTitle: 'Add Category',
      catData: { cat_name: '', cat_desc: '' },
      redirect: false,
    };
  }
  componentWillReceiveProps(props){
    this.setState({categories:props.categories})
  }

  handleShow = () =>{
    this.setState({show:true})
  }

  handleClose = () =>{
    this.setState({show:false, catData:{cat_name:'',cat_desc:'', modalTitle:'Add Category'}})
  }

  handleChange = (e) =>{
    const field = e.target.id
    let catData = this.state.catData
    catData[field] = e.target.value
    this.setState({ catData })
  }

  deleteHandler = (id, e) =>{
    this.props.deleteCategory(id)
    .then(() =>{
      toastr.success('Category Deleted Successfully')
      this.props.loadCategories(`${url}category?page=${this.props.page}`)
      let localurl = url +'myrecipes?page=1'
      this.props.loadRecipes(localurl)
    }).catch(() =>{
      toastr.error(this.props.message)
    })
  }

  handleEditData = (id) =>{
    this.props.loadCategory(`${url}category/${id}`).then(() =>{
      this.setState({
        catData:{
          cat_name:this.props.category.cat_name,
          cat_desc:this.props.category.cat_desc,
        },
        modalTitle:'Edit Category',
        categoryId:id
      })
    })
    this.setState({ show: true });
  }

  handleCategoryUpdate = (e) =>{
    e.preventDefault()
    this.props.updateCategory(this.state.categoryId,this.state.catData)
    .then(() =>{
      toastr.success('Category Updated Successfully')
      this.props.loadCategories(`${url}category?page=${this.props.page}`)
      this.handleClose()
    }).catch(xhr =>{
      toastr.error(this.props.message)
    })

  }

  addCategory = (e) =>{
    e.preventDefault()
    this.props.addCategory(`${url}category`,this.state.catData)
    .then(() =>{
      toastr.success('Category Created Successfully')
      this.props.loadCategories(`${url}category?page=1`)
      this.handleClose()
    }).catch(xhr =>{
      toastr.error(this.props.message)
      this.setState({ redirect: true });
    })
  }

  //Handle the link to the next page from pagination
  nextPage = (props) =>{
    if (props.has_next){
      this.props.loadCategories(props.next_page)
      .catch(xhr=>{
        this.setState({ redirect: true });
      })
    }
  }
  
  // Handle the link to the previous page from pagination
  previousPage =(props) =>{
      if(props.has_prev){
        this.props.loadCategories(props.previous_page)
        .catch(xhr=>{
          this.setState({ redirect: true });
        })
      }
  };

  // Handle the search functionality
  handleCategorySearch = (event) => {
    event.preventDefault();
    this.setState({showMessage:false,q: event.target.value,page:1});
    if(this.state.q){
        var updatedList = this.props.categories;
        updatedList = updatedList.filter(function(item){
          return item.cat_name.toLowerCase().search(
            event.target.value.toLowerCase()) !== -1;
        });
        this.setState({categories: updatedList});
    if(this.state.categories.length===0){
        this.setState({
            showMessage:true
        })
    }
  }
}
  render() {
    const {catData,show, modalTitle, categories, q} = this.state
    return (
      <div className="CategoriesTable col-md-9">
        <h3>&nbsp;<a onClick={this.handleShow} className="btn btn-success pull-right" id="addCategory"> Add Category</a>
        <CategoryModal handleClose={this.handleClose} catData = {catData} show={show} message={this.props.message} modalTitle={modalTitle} handleChange={this.handleChange} handleCategoryUpdate={this.handleCategoryUpdate} addCategory={this.addCategory}/>
          <div className="col-xs-7 col-sm-4 pull-right">
            <div className="input-group mb-2 mb-sm-0">
              <div className="input-group-addon">Search</div>
              <input type="text" className="form-control" id="catSearch" onChange={this.handleCategorySearch} onKeyUp={this.handleCategorySearch} placeholder="Enter your search key words here!" />
            </div>
          </div>
        </h3><br />
        {this.props.loading
          ? <center id="loader"><i className="fa fa-spinner fa-pulse fa-4x fa-fw" /></center>
          : <div>
            {categories.length > 0
              ? <div>
                <Table className="table table-striped responsive">
                  <tbody id="tbody">
                    <tr>
                      <th>ID</th>
                      <th>Title</th>
                      <th>Category</th>
                      <th>Author</th>
                      <th>Date</th>
                      <th>Actions</th>
                    </tr>

                    {!q?this.props.categories.map((category, index) =>
                      <CategoryTableRows id={index + 1} key={category.cat_id}{...category} deleteHandler={this.deleteHandler} handleEditData={this.handleEditData} />)
                    :categories.map((category, index) =>
                    <CategoryTableRows id={index + 1} key={category.cat_id}{...category} deleteHandler={this.deleteHandler} handleEditData={this.handleEditData} />)
                    }
                  </tbody>
                </Table>
                <Pagination categoryData={this.props} nextPage={this.nextPage} previousPage={this.previousPage} />
              </div>

              : <center>
                  <div className="alert alert-info col-sm-8" id="no-categories">
                  <p>No categories at the moment!<a onClick={this.handleShow} className="btn btn-primary pull-right"> Add Category</a></p>
                  </div>
                </center>
            }
            </div>
        }
      </div>
    );
  }
}
export const mapStateToProps = (state, ownProps) => {
  let disableNext = '';
  let disablePrevious = '';
  if (state.categories.has_next) {
    disableNext = 'page-item';
  } else {
    disableNext = 'page-item disabled';
  }
  if (state.categories.has_prev) {
    disablePrevious = 'page-item';
  } else {
    disablePrevious = 'page-item disabled';
  }

  return {
    categories: state.categories.categories,
    category:state.categories.category,
    page: state.categories.page,
    total_pages: state.categories.total_pages,
    has_next: state.categories.has_next,
    has_prev: state.categories.has_prev,
    previous_page: state.categories.previous_page,
    next_page: state.categories.next_page,
    message: state.categories.message,
    disablePrevious,
    disableNext,
    loading: state.ajaxCallsInProgress > 0,
  };
};
export const mapDispatchToProps = dispatch => ({

  loadCategories: url => dispatch(loadCategories(url)),
  loadCategory: url => dispatch(loadCategory(url)),
  addCategory: (url,data) => dispatch(addCategory(url, data)),
  updateCategory: (id,data) => dispatch(updateCategory(id, data)),
  deleteCategory: (id) => dispatch(deleteCategory(id)),
  loadRecipes:(url) => dispatch(loadRecipes(url))    
});

export default connect(mapStateToProps, mapDispatchToProps)(CategoriesTable);
