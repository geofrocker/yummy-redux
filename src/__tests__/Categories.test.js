import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { CategoriesTable } from '../components/dashboard/categoriesTab/CategoriesTable';
import toJson from 'enzyme-to-json';

describe('the category Table component', () => {
  const props = {
    str: {},
    history: {
      push: () => {},
    },
    categories: [{ cat_id: 1, cat_name: 'lorem', cat_desc: 'Ipsum' }],
    catData: [{ cat_id: 1, cat_name: 'lorem', cat_desc: 'Ipsum' }],
    recipes: [{ recipe_id: 1, title: 'Lorem', ingredients: 'Ipsum' }],
    loadRecipes: () => Promise.resolve(),
    loadCategories: () => Promise.resolve(),
    deleteCategory: () => Promise.resolve(),

  };
  const wrapper = mount(<CategoriesTable {...props} />);
  wrapper.setState({ categories: [{ cat_id: 1, cat_name: 'lorem' }] });
  const preventDefault = jest.fn();

  it('renders the Category table class', () => {
    expect(wrapper.find('.CategoriesTable')).toHaveLength(1);
  });

  it('renders the category Modal', () => {
    expect(wrapper.find('Modal')).toHaveLength(2);
  });

  it('renders the category Pagination', () => {
    expect(wrapper.find('Pagination')).toHaveLength(1);
  });

  it('renders seven div jsx elements', () => {
    expect(wrapper.find('div')).toHaveLength(7);
  });

  it('renders category pagination', () => {
    wrapper.setState({ data: [{ a: 'general' }] });
    expect(wrapper.find('.pagination')).toHaveLength(1);
    expect(wrapper.find('.page-link')).toHaveLength(3);
  });

  it(' handles clicks ', () => {
    expect(wrapper.find('#catPrevPage').simulate('click'));
    expect(wrapper.find('#catNextPage').simulate('click'));
    expect(wrapper.find('#catSearch').simulate('click'));
    expect(wrapper.find('#catDelete').simulate('click'));
  });
});
