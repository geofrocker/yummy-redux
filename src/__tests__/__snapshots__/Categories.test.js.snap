// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the category Table component  handle deleting the category  1`] = `
<CategoriesTable
  catData={
    Array [
      Object {
        "cat_desc": "Ipsum",
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  categories={
    Array [
      Object {
        "cat_desc": "Ipsum",
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  deleteCategory={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loadCategories={[Function]}
  loadRecipes={[Function]}
  recipes={
    Array [
      Object {
        "ingredients": "Ipsum",
        "recipe_id": 1,
        "title": "Lorem",
      },
    ]
  }
  str={Object {}}
>
  <div
    className="CategoriesTable col-md-9"
  >
    <h3>
       
      <a
        className="btn btn-success pull-right"
        id="addCategory"
        onClick={[Function]}
      >
         Add Category
      </a>
      <CategoryModal
        addCategory={[Function]}
        catData={
          Object {
            "cat_desc": "",
            "cat_name": "",
          }
        }
        handleCategoryUpdate={[Function]}
        handleChange={[Function]}
        handleClose={[Function]}
        modalTitle="Add Category"
        show={false}
      >
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropClassName="modal-backdrop"
            backdropTransition={[Function]}
            containerClassName="modal-open"
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onEntering={[Function]}
            onExited={[Function]}
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
            transition={[Function]}
          />
        </Modal>
      </CategoryModal>
      <div
        className="col-xs-7 col-sm-4 pull-right"
      >
        <div
          className="input-group mb-2 mb-sm-0"
        >
          <div
            className="input-group-addon"
          >
            Search
          </div>
          <input
            className="form-control"
            id="catSearch"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Enter your search key words here!"
            type="text"
          />
        </div>
      </div>
    </h3>
    <br />
    <div>
      <div>
        <Table
          bordered={false}
          bsClass="table"
          className="table table-striped responsive"
          condensed={false}
          hover={false}
          responsive={false}
          striped={false}
        >
          <table
            className="table table-striped responsive table"
          >
            <tbody
              id="tbody"
            >
              <tr>
                <th>
                  ID
                </th>
                <th>
                  Title
                </th>
                <th>
                  Category
                </th>
                <th>
                  Author
                </th>
                <th>
                  Date
                </th>
                <th>
                  Actions
                </th>
              </tr>
              <RecipeTableRows
                cat_desc="Ipsum"
                cat_id={1}
                cat_name="lorem"
                deleteHandler={[Function]}
                handleEditData={[Function]}
                id={1}
                key="1"
              >
                <tr>
                  <td>
                    1
                  </td>
                  <td>
                    lorem
                  </td>
                  <td>
                    Ipsum...
                  </td>
                  <td />
                  <td>
                    <t
                      ago={false}
                      calendar={false}
                      decimal={false}
                      element="time"
                      filter={[Function]}
                      format="DD/MM/YYYY"
                      fromNow={false}
                      interval={60000}
                      onChange={[Function]}
                      toNow={false}
                      unit={null}
                      unix={false}
                      utc={false}
                    >
                      <time
                        dateTime={"2018-02-18T15:58:16.853Z"}
                      >
                        18/02/2018
                      </time>
                    </t>
                  </td>
                  <td>
                    <button
                      className="btn btn-primary btn-xs"
                      id="catEdit"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-edit"
                      >
                         Edit
                      </span>
                    </button>
                     
                    <button
                      className="btn btn-danger btn-xs"
                      id="catDelete"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-trash"
                      >
                         Delete
                      </span>
                    </button>
                  </td>
                </tr>
              </RecipeTableRows>
            </tbody>
          </table>
        </Table>
        <Pagination
          categoryData={
            Object {
              "catData": Array [
                Object {
                  "cat_desc": "Ipsum",
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "categories": Array [
                Object {
                  "cat_desc": "Ipsum",
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "deleteCategory": [Function],
              "history": Object {
                "push": [Function],
              },
              "loadCategories": [Function],
              "loadRecipes": [Function],
              "recipes": Array [
                Object {
                  "ingredients": "Ipsum",
                  "recipe_id": 1,
                  "title": "Lorem",
                },
              ],
              "str": Object {},
            }
          }
          nextPage={[Function]}
          previousPage={[Function]}
        >
          <center>
            <div
              className="col-xs-12 col-sm-12"
              id="pagination"
            >
              <ul
                className="pagination"
              >
                <li>
                  <a
                    className="page-link"
                    id="catPrevPage"
                    onClick={[Function]}
                  >
                    Previous
                  </a>
                </li>
                <li>
                  <a
                    className="page-link"
                    id="catNextPage"
                    onClick={[Function]}
                  >
                    Next
                  </a>
                  <a
                    className="page-link"
                  >
                    Showing 
                     of 
                  </a>
                </li>
              </ul>
            </div>
          </center>
        </Pagination>
      </div>
    </div>
  </div>
</CategoriesTable>
`;

exports[`the category Table component  handle going back to  category previous page  1`] = `
<CategoriesTable
  catData={
    Array [
      Object {
        "cat_desc": "Ipsum",
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  categories={
    Array [
      Object {
        "cat_desc": "Ipsum",
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  deleteCategory={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loadCategories={[Function]}
  loadRecipes={[Function]}
  recipes={
    Array [
      Object {
        "ingredients": "Ipsum",
        "recipe_id": 1,
        "title": "Lorem",
      },
    ]
  }
  str={Object {}}
>
  <div
    className="CategoriesTable col-md-9"
  >
    <h3>
       
      <a
        className="btn btn-success pull-right"
        id="addCategory"
        onClick={[Function]}
      >
         Add Category
      </a>
      <CategoryModal
        addCategory={[Function]}
        catData={
          Object {
            "cat_desc": "",
            "cat_name": "",
          }
        }
        handleCategoryUpdate={[Function]}
        handleChange={[Function]}
        handleClose={[Function]}
        modalTitle="Add Category"
        show={false}
      >
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropClassName="modal-backdrop"
            backdropTransition={[Function]}
            containerClassName="modal-open"
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onEntering={[Function]}
            onExited={[Function]}
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
            transition={[Function]}
          />
        </Modal>
      </CategoryModal>
      <div
        className="col-xs-7 col-sm-4 pull-right"
      >
        <div
          className="input-group mb-2 mb-sm-0"
        >
          <div
            className="input-group-addon"
          >
            Search
          </div>
          <input
            className="form-control"
            id="catSearch"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Enter your search key words here!"
            type="text"
          />
        </div>
      </div>
    </h3>
    <br />
    <div>
      <div>
        <Table
          bordered={false}
          bsClass="table"
          className="table table-striped responsive"
          condensed={false}
          hover={false}
          responsive={false}
          striped={false}
        >
          <table
            className="table table-striped responsive table"
          >
            <tbody
              id="tbody"
            >
              <tr>
                <th>
                  ID
                </th>
                <th>
                  Title
                </th>
                <th>
                  Category
                </th>
                <th>
                  Author
                </th>
                <th>
                  Date
                </th>
                <th>
                  Actions
                </th>
              </tr>
              <RecipeTableRows
                cat_desc="Ipsum"
                cat_id={1}
                cat_name="lorem"
                deleteHandler={[Function]}
                handleEditData={[Function]}
                id={1}
                key="1"
              >
                <tr>
                  <td>
                    1
                  </td>
                  <td>
                    lorem
                  </td>
                  <td>
                    Ipsum...
                  </td>
                  <td />
                  <td>
                    <t
                      ago={false}
                      calendar={false}
                      decimal={false}
                      element="time"
                      filter={[Function]}
                      format="DD/MM/YYYY"
                      fromNow={false}
                      interval={60000}
                      onChange={[Function]}
                      toNow={false}
                      unit={null}
                      unix={false}
                      utc={false}
                    >
                      <time
                        dateTime={"2018-02-18T15:58:16.853Z"}
                      >
                        18/02/2018
                      </time>
                    </t>
                  </td>
                  <td>
                    <button
                      className="btn btn-primary btn-xs"
                      id="catEdit"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-edit"
                      >
                         Edit
                      </span>
                    </button>
                     
                    <button
                      className="btn btn-danger btn-xs"
                      id="catDelete"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-trash"
                      >
                         Delete
                      </span>
                    </button>
                  </td>
                </tr>
              </RecipeTableRows>
            </tbody>
          </table>
        </Table>
        <Pagination
          categoryData={
            Object {
              "catData": Array [
                Object {
                  "cat_desc": "Ipsum",
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "categories": Array [
                Object {
                  "cat_desc": "Ipsum",
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "deleteCategory": [Function],
              "history": Object {
                "push": [Function],
              },
              "loadCategories": [Function],
              "loadRecipes": [Function],
              "recipes": Array [
                Object {
                  "ingredients": "Ipsum",
                  "recipe_id": 1,
                  "title": "Lorem",
                },
              ],
              "str": Object {},
            }
          }
          nextPage={[Function]}
          previousPage={[Function]}
        >
          <center>
            <div
              className="col-xs-12 col-sm-12"
              id="pagination"
            >
              <ul
                className="pagination"
              >
                <li>
                  <a
                    className="page-link"
                    id="catPrevPage"
                    onClick={[Function]}
                  >
                    Previous
                  </a>
                </li>
                <li>
                  <a
                    className="page-link"
                    id="catNextPage"
                    onClick={[Function]}
                  >
                    Next
                  </a>
                  <a
                    className="page-link"
                  >
                    Showing 
                     of 
                  </a>
                </li>
              </ul>
            </div>
          </center>
        </Pagination>
      </div>
    </div>
  </div>
</CategoriesTable>
`;

exports[`the category Table component  handle going back to next category page  1`] = `
<CategoriesTable
  catData={
    Array [
      Object {
        "cat_desc": "Ipsum",
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  categories={
    Array [
      Object {
        "cat_desc": "Ipsum",
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  deleteCategory={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loadCategories={[Function]}
  loadRecipes={[Function]}
  recipes={
    Array [
      Object {
        "ingredients": "Ipsum",
        "recipe_id": 1,
        "title": "Lorem",
      },
    ]
  }
  str={Object {}}
>
  <div
    className="CategoriesTable col-md-9"
  >
    <h3>
       
      <a
        className="btn btn-success pull-right"
        id="addCategory"
        onClick={[Function]}
      >
         Add Category
      </a>
      <CategoryModal
        addCategory={[Function]}
        catData={
          Object {
            "cat_desc": "",
            "cat_name": "",
          }
        }
        handleCategoryUpdate={[Function]}
        handleChange={[Function]}
        handleClose={[Function]}
        modalTitle="Add Category"
        show={false}
      >
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropClassName="modal-backdrop"
            backdropTransition={[Function]}
            containerClassName="modal-open"
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onEntering={[Function]}
            onExited={[Function]}
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
            transition={[Function]}
          />
        </Modal>
      </CategoryModal>
      <div
        className="col-xs-7 col-sm-4 pull-right"
      >
        <div
          className="input-group mb-2 mb-sm-0"
        >
          <div
            className="input-group-addon"
          >
            Search
          </div>
          <input
            className="form-control"
            id="catSearch"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Enter your search key words here!"
            type="text"
          />
        </div>
      </div>
    </h3>
    <br />
    <div>
      <div>
        <Table
          bordered={false}
          bsClass="table"
          className="table table-striped responsive"
          condensed={false}
          hover={false}
          responsive={false}
          striped={false}
        >
          <table
            className="table table-striped responsive table"
          >
            <tbody
              id="tbody"
            >
              <tr>
                <th>
                  ID
                </th>
                <th>
                  Title
                </th>
                <th>
                  Category
                </th>
                <th>
                  Author
                </th>
                <th>
                  Date
                </th>
                <th>
                  Actions
                </th>
              </tr>
              <RecipeTableRows
                cat_desc="Ipsum"
                cat_id={1}
                cat_name="lorem"
                deleteHandler={[Function]}
                handleEditData={[Function]}
                id={1}
                key="1"
              >
                <tr>
                  <td>
                    1
                  </td>
                  <td>
                    lorem
                  </td>
                  <td>
                    Ipsum...
                  </td>
                  <td />
                  <td>
                    <t
                      ago={false}
                      calendar={false}
                      decimal={false}
                      element="time"
                      filter={[Function]}
                      format="DD/MM/YYYY"
                      fromNow={false}
                      interval={60000}
                      onChange={[Function]}
                      toNow={false}
                      unit={null}
                      unix={false}
                      utc={false}
                    >
                      <time
                        dateTime={"2018-02-18T15:58:16.853Z"}
                      >
                        18/02/2018
                      </time>
                    </t>
                  </td>
                  <td>
                    <button
                      className="btn btn-primary btn-xs"
                      id="catEdit"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-edit"
                      >
                         Edit
                      </span>
                    </button>
                     
                    <button
                      className="btn btn-danger btn-xs"
                      id="catDelete"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-trash"
                      >
                         Delete
                      </span>
                    </button>
                  </td>
                </tr>
              </RecipeTableRows>
            </tbody>
          </table>
        </Table>
        <Pagination
          categoryData={
            Object {
              "catData": Array [
                Object {
                  "cat_desc": "Ipsum",
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "categories": Array [
                Object {
                  "cat_desc": "Ipsum",
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "deleteCategory": [Function],
              "history": Object {
                "push": [Function],
              },
              "loadCategories": [Function],
              "loadRecipes": [Function],
              "recipes": Array [
                Object {
                  "ingredients": "Ipsum",
                  "recipe_id": 1,
                  "title": "Lorem",
                },
              ],
              "str": Object {},
            }
          }
          nextPage={[Function]}
          previousPage={[Function]}
        >
          <center>
            <div
              className="col-xs-12 col-sm-12"
              id="pagination"
            >
              <ul
                className="pagination"
              >
                <li>
                  <a
                    className="page-link"
                    id="catPrevPage"
                    onClick={[Function]}
                  >
                    Previous
                  </a>
                </li>
                <li>
                  <a
                    className="page-link"
                    id="catNextPage"
                    onClick={[Function]}
                  >
                    Next
                  </a>
                  <a
                    className="page-link"
                  >
                    Showing 
                     of 
                  </a>
                </li>
              </ul>
            </div>
          </center>
        </Pagination>
      </div>
    </div>
  </div>
</CategoriesTable>
`;

exports[`the category Table component  handles category search feature  1`] = `
<CategoriesTable
  catData={
    Array [
      Object {
        "cat_desc": "Ipsum",
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  categories={
    Array [
      Object {
        "cat_desc": "Ipsum",
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  deleteCategory={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loadCategories={[Function]}
  loadRecipes={[Function]}
  recipes={
    Array [
      Object {
        "ingredients": "Ipsum",
        "recipe_id": 1,
        "title": "Lorem",
      },
    ]
  }
  str={Object {}}
>
  <div
    className="CategoriesTable col-md-9"
  >
    <h3>
       
      <a
        className="btn btn-success pull-right"
        id="addCategory"
        onClick={[Function]}
      >
         Add Category
      </a>
      <CategoryModal
        addCategory={[Function]}
        catData={
          Object {
            "cat_desc": "",
            "cat_name": "",
          }
        }
        handleCategoryUpdate={[Function]}
        handleChange={[Function]}
        handleClose={[Function]}
        modalTitle="Add Category"
        show={false}
      >
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropClassName="modal-backdrop"
            backdropTransition={[Function]}
            containerClassName="modal-open"
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onEntering={[Function]}
            onExited={[Function]}
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
            transition={[Function]}
          />
        </Modal>
      </CategoryModal>
      <div
        className="col-xs-7 col-sm-4 pull-right"
      >
        <div
          className="input-group mb-2 mb-sm-0"
        >
          <div
            className="input-group-addon"
          >
            Search
          </div>
          <input
            className="form-control"
            id="catSearch"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Enter your search key words here!"
            type="text"
          />
        </div>
      </div>
    </h3>
    <br />
    <div>
      <div>
        <Table
          bordered={false}
          bsClass="table"
          className="table table-striped responsive"
          condensed={false}
          hover={false}
          responsive={false}
          striped={false}
        >
          <table
            className="table table-striped responsive table"
          >
            <tbody
              id="tbody"
            >
              <tr>
                <th>
                  ID
                </th>
                <th>
                  Title
                </th>
                <th>
                  Category
                </th>
                <th>
                  Author
                </th>
                <th>
                  Date
                </th>
                <th>
                  Actions
                </th>
              </tr>
              <RecipeTableRows
                cat_desc="Ipsum"
                cat_id={1}
                cat_name="lorem"
                deleteHandler={[Function]}
                handleEditData={[Function]}
                id={1}
                key="1"
              >
                <tr>
                  <td>
                    1
                  </td>
                  <td>
                    lorem
                  </td>
                  <td>
                    Ipsum...
                  </td>
                  <td />
                  <td>
                    <t
                      ago={false}
                      calendar={false}
                      decimal={false}
                      element="time"
                      filter={[Function]}
                      format="DD/MM/YYYY"
                      fromNow={false}
                      interval={60000}
                      onChange={[Function]}
                      toNow={false}
                      unit={null}
                      unix={false}
                      utc={false}
                    >
                      <time
                        dateTime={"2018-02-18T15:58:16.853Z"}
                      >
                        18/02/2018
                      </time>
                    </t>
                  </td>
                  <td>
                    <button
                      className="btn btn-primary btn-xs"
                      id="catEdit"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-edit"
                      >
                         Edit
                      </span>
                    </button>
                     
                    <button
                      className="btn btn-danger btn-xs"
                      id="catDelete"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-trash"
                      >
                         Delete
                      </span>
                    </button>
                  </td>
                </tr>
              </RecipeTableRows>
            </tbody>
          </table>
        </Table>
        <Pagination
          categoryData={
            Object {
              "catData": Array [
                Object {
                  "cat_desc": "Ipsum",
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "categories": Array [
                Object {
                  "cat_desc": "Ipsum",
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "deleteCategory": [Function],
              "history": Object {
                "push": [Function],
              },
              "loadCategories": [Function],
              "loadRecipes": [Function],
              "recipes": Array [
                Object {
                  "ingredients": "Ipsum",
                  "recipe_id": 1,
                  "title": "Lorem",
                },
              ],
              "str": Object {},
            }
          }
          nextPage={[Function]}
          previousPage={[Function]}
        >
          <center>
            <div
              className="col-xs-12 col-sm-12"
              id="pagination"
            >
              <ul
                className="pagination"
              >
                <li>
                  <a
                    className="page-link"
                    id="catPrevPage"
                    onClick={[Function]}
                  >
                    Previous
                  </a>
                </li>
                <li>
                  <a
                    className="page-link"
                    id="catNextPage"
                    onClick={[Function]}
                  >
                    Next
                  </a>
                  <a
                    className="page-link"
                  >
                    Showing 
                     of 
                  </a>
                </li>
              </ul>
            </div>
          </center>
        </Pagination>
      </div>
    </div>
  </div>
</CategoriesTable>
`;
